### ISSUE TITLE
Evolving any ignores branches before a 'continue'

### ISSUE THREAD

**Issue by jared-hughes** [microsoft/TypeScript#60945]
### 🔎 Search Terms

"evolving any", "implicit any", "for loop continue", "incorrect inference"

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about "evolving let"

### ⏯ Playground Link

Playground link with relevant code: https://www.typescriptlang.org/play/?ts=5.7.2#code/DYUwLgBAlgciAeYDcAoAZgewE4QBSkiggF4IAGJaCAHggEYKrTYEwBKCAbxQl4gHp+fYSNHCAegH4efKGjxFipAKwduY6HEQkIAcgBGAQwAmu1BoDGGAHZgo1gK4hzfAL4zeLbcwgBqeqiuQA

### 💻 Code

```ts
let iNext;
for (let i = 0; i < 10; i = iNext) {
    //                      ^? let iNext: number
    if (i == 5) {
        iNext = 'bad';
        continue;
    }
    iNext = i + 1;
}
```

### 🙁 Actual behavior

The type of `iNext` is an evolving any. It evolved to `number` (at the point of `i = iNext`), which is incorrect because `iNext = 'bad'` can assign it a string. The assignment before `continue` doesn't get included in the type.

### 🙂 Expected behavior

The type of `iNext` should evolve to `string | number` as measured at `i = iNext` because either the assignment `iNext = 'bad'` or `iNext = i + 1` are on a control flow path between `let iNext` and `i = iNext`.

This should give an error because `i` has type `number`.

**Comment by Andarist** [comment]
The evolving any here is a red herring. The same thing happens with an annotated variable. It's some general issue with CFA: [TS playground](https://www.typescriptlang.org/play/?ts=5.7.2#code/DYUwLgBAlgciAeYBcEDOYBOUB2BzCAPhNgK4C2ARiBhALwQBEDA3AFABmA9jQBSiRQ6EAAzNoEADwQAjKPH1YCMAEoIAb1YQIAem1b9Bw0a0A9APybo7CD0G16AVlUaDixEIYUAhgBMWrSy0AY05sMBwSEDYtAF9LN0gFCABqGTY4oA)
